src/index.ts

```diff
@@ -174,16 +178,52 @@ export class MayanRoute<N extends Network>
     };
 
     const quotes = await fetchQuote(quoteOpts);
-    if (quotes.length === 0) throw new Error("Failed to find quotes");
+    if (quotes.length === 0) return undefined;
+    if (quotes.length === 1) return quotes[0];
+
+    // Wormhole SDK routes return only a single quote, but Mayan offers multiple quotes (because 
+    // Mayan comprises multiple competing protocols). We sort the quotes Mayan gives us and choose
+    // the best one here.
+    //
+    // User can provide optimizeFor option to indicate what they care about. It defaults to "cost"
+    // which just optimizes for highest amount out, but it can also be set to "speed" which will
+    // choose the fastest route instead.
+    quotes.sort((a: MayanQuote, b: MayanQuote) => {
+      if (params.options.optimizeFor === 'cost') {
+        if (b.expectedAmountOut === a.expectedAmountOut) {
+          // If expected amounts out are identical, fall back to speed
+          return a.eta - b.eta
```

Comment:
Should we use etaSeconds?

Reply:
yeah maybe that's better. typescript will complain until this is merged but thats ok i'll just use the magic words https://github.com/mayan-finance/swap-sdk/pull/11